core/templates/domain/statistics/PlaythroughIssueObjectFactorySpec.ts

```diff
@@ -16,47 +16,85 @@
  * @fileoverview Unit tests for the PlaythroughIssueObjectFactory.
  */
 
-import { PlaythroughIssueObjectFactory, PlaythroughIssue } from
-  'domain/statistics/PlaythroughIssueObjectFactory';
+import {
+  IPlaythroughIssueBackendDict,
+  PlaythroughIssueObjectFactory,
+  PlaythroughIssue
+} from 'domain/statistics/PlaythroughIssueObjectFactory';
 
 describe('Playthrough Issue Object Factory', () => {
   let piof: PlaythroughIssueObjectFactory;
-  let playthroughIssueObject: PlaythroughIssue;
+  let playthroughIssueObject: PlaythroughIssue<'EarlyQuit'>;
```

Comment:
Don't do this. This would make it unusable in AngularJS files, which haven't migrated to Angular 8 yet and need to import services in the "old" style (which wouldn't support generic type parameters b/c of the lack of type-support.

Reply:
Umm, actually I think we can still use it in the AngularJS files normally without the parameter.

Reply:
Even so, let's not expose a generic function. Please only expose concrete types and concrete functions, like everything else in the codebase.

Reply:
Not done? Please change before reassigning.

Reply:
Actually wanted you to see https://github.com/oppia/oppia/pull/9596#discussion_r443750299

Reply:
Gotcha, done!

Reply:
done